applications/ShapeOptimizationApplication/custom_utilities/response_functions/face_angle_response_function_utility.cpp

```diff
@@ -42,13 +43,35 @@ FaceAngleResponseFunctionUtility::FaceAngleResponseFunctionUtility(ModelPart& rM
 	}
 	else
 		KRATOS_ERROR << "Specified gradient_mode '" << gradient_mode << "' not recognized. The only option is: finite_differencing" << std::endl;
+
+	mConsiderOnlyInitiallyFeasible = ResponseSettings["consider_only_initially_feasible"].GetBool();
+}
+
+void FaceAngleResponseFunctionUtility::Initialize()
+{
+	KRATOS_TRY;
+
+	if (!mConsiderOnlyInitiallyFeasible){
+		return;
+	}
+
+    KRATOS_INFO("ShapeOpt") << "Considering only initially feasible faces for face angle response."<< std::endl;
+	block_for_each(mrModelPart.Conditions(), [&](Condition& rCond) {
+		const double g_i = CalculateConditionValue(rCond);
+		rCond.SetValue(FACE_ANGLE_CONSIDER, (g_i <= 0));
```

Comment:
just a thought, Wouldn't it be better if the variable name is `CONSIDER_FACE_ANGLE` since it is a bool?

Reply:
I have renamed it, can you please re-approve? Thanks!